plugin/src/main/java/com/google/errorprone/xplat/checker/XplatBans.java

```diff
@@ -0,0 +1,359 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.errorprone.xplat.checker;
+
+import static com.google.errorprone.BugPattern.SeverityLevel.ERROR;
+
+import com.google.common.io.Resources;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.ErrorProneFlags;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.MethodTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.NewClassTreeMatcher;
+import com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.NewClassTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.VariableTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import com.sun.tools.javac.code.Type;
+import com.google.common.base.Charsets;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Optional;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+
+/**
+ * Check for usage of some Joda-Time classes and packages, which can be found in
+ * resources/Xplatbans.json. These calls are banned due to their incompatibility with cross platform
+ * development. Additional classes can be banned using the command line argument {@code
+ * -XepOpt:XplatBans:JSON=X}, where X is the path to a JSON file containing custom bans. Errors can
+ * be suppressed with the {@link XplatBanSuppression} annotation.
+ */
+@BugPattern(
+    name = "XplatBans",
+    summary = "Bans the usage of certain Joda-Time classes and packages for cross platform use.",
+    explanation =
+        "The usage of several Joda-Time classes and packages are banned from cross"
+            + " platform development due to incompatibilities. They are unsupported on the web"
+            + " and should also not be used on supported platforms. Additional bans can be configured"
+            + " with the command line argument -XepOpt:XplatBans:JSON=X, where X is the path to"
+            + " a JSON file containing custom bans.",
+    severity = ERROR,
```

Comment:
The text of the bug explanation should be updated to not talk specifically about Joda-Time.

Also a description of the file format would probably help.

Reply:
Changed the description and added the file format